src/NuGet.Jobs.GitHubIndexer/GitRepoSearchers/GitHubSearcher.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class GitHubSearcher : IGitRepoSearcher
+    {
+        public const int MIN_STARS = 100;
+        public const int RESULTS_PER_PAGE = 100; // Maximum is 100 :(
+        private IGitHubClient _client;
+
+        public GitHubSearcher(IGitHubClient client)
+        {
+            _client = client ?? throw new ArgumentNullException(nameof(client));
+        }
+
+        private async Task<List<Repository>> GetResultsForPage(int currPage, int totalCount, int maxStarCount = -1, string lastRecordName = null)
+        {
+            if (_client.GetLastApiInfo() != null && _client.GetLastApiInfo().RateLimit.Remaining == 0)
+            {
+                Console.WriteLine("Waiting a minute to cooldown..");
```

Comment:
Please use ASP.NET Core logging instead of `Console.WriteLine`: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.2

```suggestion
                _logger.LogInformation("Waiting a minute to cooldown..");
```